### PR TITLE
Fixes #9231 - Require English lib and fix ENC failure

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

**Comment by domcleal** [comment]
There's also the instance in app/host/managed.rb changed here: https://github.com/theforeman/foreman/commit/4f7a4d0ba84f43e3cb0d422d9f5e82b01588f21b#diff-29

(Did you mean to keep English?)

**Comment by daviddavis** [comment]
Yea, I [added the English lib](https://github.com/theforeman/foreman/pull/2131/files#diff-9a29cd62971a5dd9e4a2bd514f4a019cR11). If you'd rather I went back to the original globals, I can do that. Or if you want me to refactor 4f7a4d0#diff-29, I can do that too.

**Comment by daviddavis** [comment]
FYI, there's also other places that require English too like https://github.com/theforeman/foreman/commit/4f7a4d0ba84f43e3cb0d422d9f5e82b01588f21b#diff-53e1a0d0a8734476a8eb9b7e66b59d4aL164 (or reversion).

**Comment by domcleal** [comment]
Ah cool, that's fine by me.  I thought CHILD_STATUS was core, interesting!

File: app/controllers/hosts_controller.rb [hunk 1]
```diff
@@ -176,10 +176,11 @@ def externalNodes
...
         format.yml { render :text => @host.info.to_yaml }
       end
-    rescue
+    rescue => e
       # failed
-      logger.warn "Failed to generate external nodes for #{@host} with #{$ERROR_INFO}"
-      render :text => _('Unable to generate output, Check log files\n'), :status => 412 and return
+      logger.warn "Failed to generate external nodes for #{@host} with #{e}"
+      logger.debug(e.backtrace.join("\n"))
+      render :text => _('Unable to generate output, Check log files'), :status => 412 and return
```
Comment (dLobatog) [inline app/controllers/hosts_controller.rb]: Nitpick, can you use the symbol instead of the code? `:precondition_failed`

**Comment by dLobatog** [comment]
@daviddavis :+1: , but is this still WIP?

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
Updated. Should be ready to go.

**Comment by dLobatog** [comment]
Merged as 350a0417951ebff465cda10645eae5c1dedae2cc , thanks @daviddavis . Notice I removed the '#failed' comment as it was very redundant

**Pull request closed without merging**